### DISCUSSION TITLE
SSL Configuration Diffie-Hellman prime

### DISCUSSION THREAD

**Discussion by nestle2377** [keycloak/keycloak#28398] [Archive]
Hi to all. I've an installation of keycloak on quark distribution.  My security office during an audit has found that keycloak has this vulnerability: 
**Diffie-Hellman prime is less than 2048 bits.** 

I immagine that to configure 2048 bits is something related to quarkus environment. 

Do you know how to fix it? is there some configurations?

thanks

**Comment by codespearhead** [comment]
@nestle2377 Is your Keycloak instance running behind a reverse proxy? If so, the reverse proxy deals with TLS/SSL whereas everything behind the reverse proxy doesn't, correct?

**Reply by nestle2377** [reply]
No, is not behind a reverse proxy, but is exposed directly. 

is there a way to fix on keycloak?

**Reply by codespearhead** [reply]
@nestle2377 Can you confirm the Keycloak version you're running and that you're serving it over HTTPS? Because that might have to do with [a problem in an old Quarkus version](https://github.com/keycloak/keycloak/issues/21569) (and therefore, an old Keycloak version).

**Reply by nestle2377** [reply]
keycloak version is 21.1.2, is this affected? 
thanks!

**Reply by codespearhead** [reply]
@nestle2377 [Keycloak 21.x uses Quarkus 2.13.8](https://github.com/keycloak/keycloak/blob/6ff1c40a15b58de5cef653e7f2296473a012abcf/pom.xml#L44), which is affected by #21569 [unlike Quarkus 3.2](https://github.com/keycloak/keycloak/issues/21569#issuecomment-1653325456), which is the [first LTS release of Quarkus](https://quarkus.io/blog/lts-releases/).

(The Red Hat build of) [Keycloak 22.x](https://access.redhat.com/products/red-hat-build-of-keycloak/) is the first LTS release of this project, and [it uses Quarkus 3.2](https://github.com/keycloak/keycloak/blob/c8a561843424c65a553ccb82c2323536a3d76008/pom.xml#L48).

Hence, I suggest upgrading to at least version 22. Ideally you'd upgrade to the latest major version of Keycloak.

**Reply by nestle2377** [reply]
Thanks! All is clear! Thanks for your time.

**Reply by codespearhead** [reply]
You're welcome! If that solves your problem, can you [mark this discussion as answered](https://docs.github.com/en/discussions/managing-discussions-for-your-community/moderating-discussions#marking-a-comment-as-an-answer)?
